cmd/definitions/type.go

```diff
@@ -307,14 +308,34 @@ func (d *Data) LoadService(filePath string) {
 		ns.New.srv = srv
 		ns.New.ns = ns
 		ns.New.Implemented = true
-		for _, v := range d.PairsMap {
-			if v.Defaultable {
-				ns.New.Optional = append(ns.New.Optional, "default_"+v.Name)
+		for _, v := range ns.ParsedDefaultable() {
+			ns.New.Optional = append(ns.New.Optional, "default_"+v.Pair.Name)
+		}
+		// add default_*_pairs and *_features
+		hasDefaultNsPairs := false
+		hasNsFeatures := false
+		defaultNsPairsName := fmt.Sprintf("default_%s_pairs", ns.Name)
+		nsFeaturesName := fmt.Sprintf("%s_features", ns.Name)
+		for _, v := range ns.New.Optional {
+			if v == defaultNsPairsName {
+				hasDefaultNsPairs = true
```

Comment:
How about adding a warning here to let maintain remove the `default_storage_pairs` and `storage_features`? As we will auto generate for them.

Reply:
Maybe I also should remove the pairs from tests.